src/main/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraService.scala

```diff
@@ -0,0 +1,75 @@
+
+package org.broadinstitute.dsde.agora.server.webservice
+
+import org.broadinstitute.dsde.agora.server.model.{AgoraEntityType, AgoraEntity, AgoraEntityProjection}
+import org.broadinstitute.dsde.agora.server.webservice.util.ServiceMessages
+import org.broadinstitute.dsde.agora.server.webservice.validation.{AgoraValidation, AgoraValidationRejection}
+import org.joda.time.DateTime
+import spray.routing.{HttpService, Route}
+
+trait AgoraService extends HttpService with PerRequestCreator with AgoraDirectives {
+  this: ServiceHandlerProps =>
+
+  private implicit val ec = actorRefFactory.dispatcher
+
+  import org.broadinstitute.dsde.agora.server.model.AgoraApiJsonSupport._
+  import spray.httpx.SprayJsonSupport._
+
+  def path: String
+  def routes = queryByNamespaceNameSnapshotIdRoute ~ queryRoute ~ postRoute
+
+  def queryByNamespaceNameSnapshotIdRoute =
+    path(path / Segment / Segment / Segment) { (namespace, name, snapshotId) =>
+      get {
+        requestContext =>
+          perRequest(requestContext, queryHandlerProps, ServiceMessages.QueryByNamespaceNameSnapshotId(requestContext, namespace, name, snapshotId.toInt,
+          AgoraEntityType.byPath(path)))
+      }
+    }
+
+  def queryRoute =
+    path(path) {
+      get {
+        parameters("namespace".?, "name".?, "snapshotId".as[Int].?,
```

Comment:
Can we pretty-print this a little more cleanly? The 'as' on EntityType and on the 'as' on the whole parameters object are kinda close together